### PR TITLE
Avyan Consulting - Ignite Pull Request (Enterprise Azure Governance and Operations Automation)

### PR DESCRIPTION
**Pull request by consultguru**
Creating a pull request for Ignite Azure Partner Quick Starts

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@consultguru__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by singhkays** [comment]
@consultguru Left some review comments. Adding @gatneil @bmoore-msft for another pass

File: azure-governance-cloudwise/CloudWiseDeploy.ps1 [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿cls
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/CloudWiseDeploy.ps1]: remove this file and ensure the sample deploys with the scripts in the root of the repo

File: azure-governance-cloudwise/Deploy-AzureResourceGroup.ps1 [hunk 2]
```diff
@@ -0,0 +1,121 @@
+﻿#Requires -Version 3.0
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/Deploy-AzureResourceGroup.ps1]: remove this copy, and deploy with the script in the root of the repo

File: azure-governance-cloudwise/OMSAzureDashboards/Archived/OMSSQLAzureMonitoringSolution.json [hunk 3]
```diff
@@ -0,0 +1,459 @@
+{
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/OMSAzureDashboards/Archived/OMSSQLAzureMonitoringSolution.json]: if these are truly "archived" just delete them.  If they're still needed as part of the sample, move to the nested folder - if they're part of a standalone sample (that's not deprecated) put them in another folder.

File: azure-governance-cloudwise/OMSAzureDashboards/Archived/SQL Azure Monitoring Solution.omssolutiondashboard [hunk 4]
```diff
@@ -0,0 +1,454 @@
+{
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/OMSAzureDashboards/Archived/SQL Azure Monitoring Solution.omssolutiondashboard]: same comment as for other archived files

File: azure-governance-cloudwise/OMSAzureDashboards/Archived/Web App Azure Monitoring Solution.omssolutiondashboard [hunk 5]
```diff
@@ -0,0 +1,295 @@
+{
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/OMSAzureDashboards/Archived/Web App Azure Monitoring Solution.omssolutiondashboard]: same comment as for other archived files

File: azure-governance-cloudwise/OMSAzureDashboards/OMSSQLDBAzureMonitoringSolution.omsview [hunk 6]
```diff
@@ -0,0 +1,489 @@
+{
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/OMSAzureDashboards/OMSSQLDBAzureMonitoringSolution.omsview]: if this is a template file, should have a .json extension.
follow the BPs for template placement (nested or a different sample)?

File: azure-governance-cloudwise/ServicePrincipal.ps1 [hunk 7]
```diff
@@ -0,0 +1,134 @@
+﻿cls
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/ServicePrincipal.ps1]: Seems like this should be part of the template itself?  Any reason why it cannot be?

File: azure-governance-cloudwise/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,573 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "skuName": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: Can we just pick an appropriate size for this workload and make it a variable?

File: azure-governance-cloudwise/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,573 @@
...
+        "P1",
+        "P2",
+        "P3",
+        "P4"
+      ],
+      "metadata": {
+        "description": "Describes plan's pricing tier and instance size. Check details at https://azure.microsoft.com/en-us/pricing/details/app-service/"
+      }
+    },
+    "skuCapacity": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: same comment as sku

File: azure-governance-cloudwise/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,573 @@
...
+      "type": "int",
+      "defaultValue": 1,
+      "minValue": 1,
+      "metadata": {
+        "description": "Describes plan's instance count"
+      }
+    },
+    "sqlServerName": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: move to variable and generate a name

File: azure-governance-cloudwise/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,573 @@
...
+        "description": "The account name to use for the database server administrator."
+      }
+    },
+    "sqlAdministratorLoginPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password to use for the database server administrator."
+      }
+    },
+    "sqlDatabaseName": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: move to variable

File: azure-governance-cloudwise/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,573 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "The name of the new database to create."
+      },
+      "defaultValue": "cloudwise"
+    },
+    "sqlCollation": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: move to variable

File: azure-governance-cloudwise/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,573 @@
...
+      "type": "string",
+      "defaultValue": "SQL_Latin1_General_CP1_CI_AS",
+      "metadata": {
+        "description": "The database collation for governing the proper use of characters."
+      }
+    },
+    "sqlEdition": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: move to variable

File: azure-governance-cloudwise/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,573 @@
...
+      "type": "string",
+      "defaultValue": "Basic",
+      "metadata": {
+        "description": "The type of database to create. The available options are: Web, Business, Basic, Standard, and Premium."
+      }
+    },
+    "sqlMaxSizeBytes": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: move to variable

File: azure-governance-cloudwise/azuredeploy.json [hunk 15]
```diff
@@ -0,0 +1,573 @@
...
+      "type": "string",
+      "defaultValue": "1073741824",
+      "metadata": {
+        "description": "The maximum size, in bytes, for the database. Default is 1GB"
+      }
+    },
+    "sqlRequestedServiceObjectiveName": {
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: move to variable

File: azure-governance-cloudwise/azuredeploy.json [hunk 16]
```diff
@@ -0,0 +1,573 @@
...
+  },
+  "resources": [
+    {
+      "name": "[variables('sqlserverName')]",
+      "type": "Microsoft.Sql/servers",
+      "location": "[resourceGroup().location]",
+      "tags": {
+        "displayName": "SqlServer"
+      },
+      "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: update to removepreview version

File: azure-governance-cloudwise/azuredeploy.json [hunk 17]
```diff
@@ -0,0 +1,573 @@
...
+      "properties": {
+        "administratorLogin": "[parameters('sqlAdministratorLogin')]",
+        "administratorLoginPassword": "[parameters('sqlAdministratorLoginPassword')]"
+      },
+      "resources": [
+        {
+          "name": "[parameters('sqlDatabaseName')]",
+          "type": "databases",
+          "location": "[resourceGroup().location]",
+          "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: update to remove preview version

File: azure-governance-cloudwise/azuredeploy.json [hunk 18]
```diff
@@ -0,0 +1,573 @@
...
+          ],
+          "properties": {
+            "edition": "[parameters('sqlEdition')]",
+            "collation": "[parameters('sqlCollation')]",
+            "maxSizeBytes": "[parameters('sqlMaxSizeBytes')]",
+            "requestedServiceObjectiveName": "[parameters('sqlRequestedServiceObjectiveName')]"
+          }
+        },
+        {
+          "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/azuredeploy.json]: update to remove preview version

File: azure-governance-cloudwise/nested/AutomationRunbooksOMSDashboard/OMSAutomationRunbooksDashboard.json [hunk 19]
```diff
@@ -0,0 +1,343 @@
...
+      "metadata": {
+        "description": "Path of the template folder."
+      }
+    },
+    "_artifactsLocationSasToken": {
+      "type": "securestring"
+    }
+  },
+  "variables": {
+    "laApiVersion": "2015-11-01-preview",
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/nested/AutomationRunbooksOMSDashboard/OMSAutomationRunbooksDashboard.json]: use literal apiVersions so schema validation can be done

File: azure-governance-cloudwise/nested/AutomationRunbooksOMSDashboard/OMSAutomationRunbooksDashboard.json [hunk 20]
```diff
@@ -0,0 +1,343 @@
...
+    "sqlIngestionScriptUri": "[concat(parameters('_artifactsLocation'),'/scripts/sqlazureIngestion.ps1')]",
+    "scheduleScriptUri": "[concat(parameters('_artifactsLocation'),'/scripts/scheduleIngestion.ps1')]",
+    "scheduleRunbookTemplateFileName": "[concat(parameters('_artifactsLocation'),'/deployRunbookOnSchedule.json')]",
+    "runonceJobSchedulerName": "runoncescheduleingestion",
+    "sku": "Basic",
+
+    "psModules": {
+      "AzureRmProfile": {
+        "Name": "AzureRm.Profile",
+        "URL": "https://devopsgallerystorage.blob.core.windows.net/packages/azurerm.profile.1.0.9.nupkg"
```
Comment (bmoore-msft) [inline azure-governance-cloudwise/nested/AutomationRunbooksOMSDashboard/OMSAutomationRunbooksDashboard.json]: do we need to hardcode version numbers or can we use "latest"?

Comment (consultguru) [inline azure-governance-cloudwise/nested/AutomationRunbooksOMSDashboard/OMSAutomationRunbooksDashboard.json]: This is not really an ARM deploy json file. I've changed the fileextension to omsview and updated the install guide

File: azure-governance-cloudwise/nested/AutomationRunbooksOMSDashboard/OMSAutomationRunbooksDashboard.json (hunk 19 above)
Comment (consultguru) [inline azure-governance-cloudwise/nested/AutomationRunbooksOMSDashboard/OMSAutomationRunbooksDashboard.json]: This is not really an ARM deploy json file. I've changed the fileextension to omsview and updated the install guide

File: azure-governance-cloudwise/azuredeploy.json (hunk 18 above)
Comment (consultguru) [inline azure-governance-cloudwise/azuredeploy.json]: Couldn't find the latest SQL DB schemas.
1) Looked @ https://github.com/Azure/azure-resource-manager-schemas
2) as well as scanned the quickstarts repository.

All projects use "2014-04-01-preview"

Please advise if there's a latest one.

File: azure-governance-cloudwise/azuredeploy.json (hunk 16 above)
Comment (consultguru) [inline azure-governance-cloudwise/azuredeploy.json]: Couldn't find the latest SQL DB schemas.
1) Looked @ https://github.com/Azure/azure-resource-manager-schemas
2) as well as scanned the quickstarts repository.

All projects use "2014-04-01-preview"

Please advise if there's a latest one.

File: azure-governance-cloudwise/azuredeploy.json (hunk 15 above)
Comment (consultguru) [inline azure-governance-cloudwise/azuredeploy.json]: done

File: azure-governance-cloudwise/azuredeploy.json (hunk 13 above)
Comment (consultguru) [inline azure-governance-cloudwise/azuredeploy.json]: done

File: azure-governance-cloudwise/azuredeploy.json (hunk 12 above)
Comment (consultguru) [inline azure-governance-cloudwise/azuredeploy.json]: done

File: azure-governance-cloudwise/azuredeploy.json (hunk 9 above)
Comment (consultguru) [inline azure-governance-cloudwise/azuredeploy.json]: That  was our original plan, however a few customers who tried this @ Ignite suggested to keep these as parameters as they plan to configure it with a slightly better skus for their dev environment.

File: azure-governance-cloudwise/azuredeploy.json (hunk 8 above)
Comment (consultguru) [inline azure-governance-cloudwise/azuredeploy.json]: That  was our original plan, however a few customers who tried this @ Ignite suggested to keep these as parameters as they plan to configure it with a slightly better skus for their dev environment.

File: azure-governance-cloudwise/ServicePrincipal.ps1 (hunk 7 above)
Comment (consultguru) [inline azure-governance-cloudwise/ServicePrincipal.ps1]: ServicePrincipals were currently not supported creation through JSON templates, so we sougnt to keep this manual for now. Documented in the install guide images/DeploymentGuide_and_Configuration_Guide_CloudWise.pdf > 1.3

File: azure-governance-cloudwise/OMSAzureDashboards/OMSSQLDBAzureMonitoringSolution.omsview (hunk 6 above)
Comment (consultguru) [inline azure-governance-cloudwise/OMSAzureDashboards/OMSSQLDBAzureMonitoringSolution.omsview]: not a template file. The OMS views need to be installed manually.Steps documented in the images/DeploymentGuide_and_Configuration_Guide_CloudWise.pdf > section 1.6

File: azure-governance-cloudwise/OMSAzureDashboards/Archived/Web App Azure Monitoring Solution.omssolutiondashboard (hunk 5 above)
Comment (consultguru) [inline azure-governance-cloudwise/OMSAzureDashboards/Archived/Web App Azure Monitoring Solution.omssolutiondashboard]: removed

File: azure-governance-cloudwise/OMSAzureDashboards/Archived/OMSSQLAzureMonitoringSolution.json (hunk 3 above)
Comment (consultguru) [inline azure-governance-cloudwise/OMSAzureDashboards/Archived/OMSSQLAzureMonitoringSolution.json]: all archived files removed

File: azure-governance-cloudwise/Deploy-AzureResourceGroup.ps1 (hunk 2 above)
Comment (consultguru) [inline azure-governance-cloudwise/Deploy-AzureResourceGroup.ps1]: done

File: azure-governance-cloudwise/CloudWiseDeploy.ps1 (hunk 1 above)
Comment (consultguru) [inline azure-governance-cloudwise/CloudWiseDeploy.ps1]: done

**Comment by consultguru** [comment]
Updated commit based on review comments

**Comment by singhkays** [comment]
@consultguru Can you resolve the merge conflicts?

**Pull request closed without merging**